nextboot/nextboot.go

```diff
@@ -0,0 +1,147 @@
+type V1 struct {
+
+	// Chain is a URL to a new nextboot configuration. The contents of
+	// that configuration may contain a new Chain URL, but should typically
+	// refer to a config with Commands.
+	Chain string `json:"chain,omitempty"`
+
+	// Kargs contains kernel command line parameters, typically read from
+	// /proc/cmdline. Kernel parameters are split on the first `=`, taking the
+	// left hand side as the Kargs key, and the right hand side as the value.
+	// If there is no `=` in the parameter, the entire parameter becomes the
+	// key with an empty value. All keys and values are strings.
+	//
+	// For example, if there were a parameter `ide-core.nodma=0.1`, then Kargs
+	// will contain a key `ide-core.nodma` with a value of `0.1`. Kargs may be
+	// referenced in templates using the `kargs` template function. See more
+	// examples below.
+	Kargs map[string]string `json:"kargs,omitempty"`
+
+	// Vars contains key/value pairs. Every string value is evaluated as
+	// a template. Every template value may only reference kernel parameters
+	// using the "kargs" template function. For example, if there was originally
+	// a kernel parameter "net.arg1=test", then a variable template like:
+	//
+	//     "this is a {{kargs `net.arg1`}}"
+	//
+	// would evaluate as and be replaced with:
+	//
+	//     "this is a test"
+	//
+	// For compatibility with template lookup, Vars keys must not contain ".".
+	//
+	// Vars supports three value types:
+	//
+	//   1. string - evaluated as a template.
+	//   2. []string -- as a convenience, an array of strings is first converted
+	//         to a single string by joining each element with a space. After
+	//         converstion the result is evaluated as a template.
+	//   3. map[string]interface{} -- as a convenience, allows creation of
+	//         recursively nested names.
+	//
+	// Other types are ignored.
+	//
+	// Vars may be empty.
+	Vars map[string]interface{} `json:"vars,omitempty"`
+
+	// Files is a map of names to source specs. Every source spec must contain a
+	// URL using the key name "url". The source spec may contain a sha256
+	// checksum of the file using the key name "sha256". After downloading the
+	// URL to a local file, the checksum is used to verify the file contents.
+	//
+	// The local filename is added to the source spec using the key "name" so
+	// that templates may access file names through the ".files" namespace.
+	//
+	// Each URL is evaluated as a template. File URLs may
+	// reference kernel parameters using the "kargs" template function, and
+	// variable values using the ".vars" namespace.
+	//
+	// For compatibility with template lookup, Files keys must not contain ".".
+	//
+	// For example, after download, the local file name of a source spec like:
+	//
+	// "files": {
+	//    "initram" : {
+	//       "url" : "https://storage.com/coreos_initram.cpio.gz",
+	//       "sha256" : "37c0e81be3a24752fcc2bc51c20e8dae897417dfaabbdce3a8b1efc8a2d310c6"
```

Comment:
Maybe add an optional `template` argument of a `mkstemp` template string?  Or did we decide that if you want a canonical name then you should do something like:
```sh
cp {{.files.initram.name}} /initram.stage2
```
in the commands?